File: python3/1790.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        s = 0
        n = len(s1)
        while s < n and s1[s] == s2[s]:
            s += 1
        if s == n:
            return True
        e = n - 1
        while s1[e] == s2[e]:
            e -= 1
        if s1[s] != s2[e] or s1[e] != s2[s]:
            return False
        s += 1
        while s < e:
            if s1[s] != s2[s]:
                return False
            s += 1
        return True

File: java/1790.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
        int s = 0, n = s1.length();
        while (s < n && s1.charAt(s) == s2.charAt(s))
            ++s;
        if (s == n)
            return true;
        int e = n - 1;
        while (s1.charAt(e) == s2.charAt(e))
            --e;
        if (s1.charAt(s) != s2.charAt(e) || s1.charAt(e) != s2.charAt(s))
            return false;
        ++s;
        while (s < e) {
            if (s1.charAt(s) != s2.charAt(s))
                return false;
            ++s;
        }
        return true;
    }
}
